Migrate to new build structure

jwest115 opened the pull request:
For #6, removed `project` calls from the `LedBlinker` deployment since the new cmake structure is to do any project calls outside of an fprime deployment and within the project root.

I looked through the documentation and didn't see any reference to `project` within deployments, so I don't think any cleanup is needed there. However, I did notice that the `project.cmake` file is missing from the `LedBlinker` deployment and that it is referenced in the [initial integration](https://github.com/fprime-community/fprime-workshop-led-blinker/blob/main/docs/initial-integration.md#adding-led-component-to-the-deployment) and [component implementation](https://github.com/fprime-community/fprime-workshop-led-blinker/blob/main/docs/component-implementation-1.md#create-the-component) docs. Do we want to add this in?

thomas-bc reviewed (commented):
Looks good! A couple things:
- let's remove the deployment's `LedBlinker/settings.ini` and leave only the one at the project root, to avoid confusion
- agreed that we may want to add in project.cmake to match the structure that `fprime-util new` generates and from the other tutorials. See https://github.com/fprime-community/fprime-tutorial-hello-world

thomas-bc commented on LedBlinker/CMakeLists.txt:
```diff
@@ -1,33 +1,21 @@
-add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../Components")
 add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/Top")
+# Add custom components to this specific deployment here
+add_fprime_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../Components")
```
This will not be needed once you add it in `project.cmake` and include the `LedBlinker` deployment in it.

jwest115 replied:
Updated based on suggestion

jwest115 commented:
Thanks for the feedback @thomas-bc! I added the `project.cmake` file to the project root and removed the `settings.ini` from the `LedBlinker` deployment. I do have one follow up question about `project.cmake` - is it now generated as part of `fprime-util new --project`?

thomas-bc commented:
> follow up question about `project.cmake` - is it now generated as part of `fprime-util new --project`?

[Yes](https://github.com/fprime-community/fprime-tools/blob/devel/src/fprime/cookiecutter_templates/cookiecutter-fprime-project/%7B%7Bcookiecutter.project_name%7D%7D/project.cmake)! It only contains the templated `Components/` directory at first, and when you run something like `fprime-util new --deployment`, it'll stick the deployment in there as well.

thomas-bc reviewed (approved):
LGTM!
